common/strings.c

```diff
@@ -175,6 +175,56 @@ TDNFSplitStringToArray(
     goto cleanup;
 }
 
+uint32_t
+TDNFJoinArrayToString(
+    char **ppszArray,
+    char *pszSep,
+    int count,
+    char **ppszResult
+)
+{
+    uint32_t dwError = 0;
+    int i, p = 0;
+    int nSize = 0, nSepSize = 0;
+    char *pszResult = NULL;
+
+    if(!ppszArray || !pszSep || !ppszResult)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    nSepSize = strlen(pszSep);
+    nSize = nSepSize * count;
+    for(i = 0; i < count && ppszArray[i]; i++)
+    {
+        nSize += strlen(ppszArray[i]);
+    }
+    nSize++;
```

Comment:
I am assuming this increment for extra null byte. Please correct me if i am wrong

Reply:
Yes.